packages/framework/src/Model/Product/Parameter/ParameterFacade.php

```diff
@@ -218,7 +219,7 @@ public function editParameterValue(int $parameterValueId, ParameterValueData $pa
         }
 
         if (count($parameterValueData->colourIcon->uploadedFilenames) === 0 && $parameterValueData->colourIcon->filesToDelete) {
-            $this->uploadedFileFacade->deleteAllUploadedFilesByEntity($parameterValue);
+            $this->uploadedFileFacade->deleteFiles($parameterValue, $parameterValueData->colourIcon->filesToDelete, UploadedFileTypeConfig::DEFAULT_TYPE_NAME);
```

Comment:
That's correct. It looks like it was not handled correctly previously. If there was another file type for the parameter value in the previous solution and I wanted to change only `colourIcon`, it would delete all the files, even for the other file types.